cmd/builder/gen/controllers.go

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gen
+
+import (
+	"fmt"
+	"strings"
+
+	j "github.com/dave/jennifer/jen"
```

Comment:
this is fine, but I think using 'text/template' for this task is better.

Reply:
`jennifer` can do some checks. And can split the generated files into pieces. Emmm, it's a trade off choice.